queries/nix/highlights.scm

```diff
@@ -62,33 +65,122 @@
   universal: (identifier) @parameter
   ":" @punctuation.special)
 
+; function calls
+(apply_expression
+  function: (variable_expression
+    name: (identifier) @function.call))
+
 ; basic identifiers
 (variable_expression) @variable
 
 ; builtin functions
-((identifier) @_i (#match? @_i "^(builtins|baseNameOf|dirOf|fetchTarball|map|removeAttrs|toString)$")) @variable.builtin
-
-; display entire builtins path as builtin (ex. `builtins.filter` is highlighted as one long builtin)
-(select_expression
-  expression: ((variable_expression) @_i (#eq? @_i "builtins"))
-  attrpath: (attrpath attr: (identifier) @variable.builtin)) @variable.builtin
-
-; import
-((identifier) @_i (#eq? @_i "import")) @include
-
-; null
-((identifier) @_i (#eq? @_i "import")) @constant.builtin
-
-; these are technically functions but they act more like keywords (abort and throw are control flow, derivation is a core language construct)
-((identifier) @_i (#match? @_i "^(abort|derivation|throw)$")) @keyword
+; modified from `nix __dump-builtins | jq keys`
+((identifier) @function.builtin (#any-of? @function.builtin
```

Comment:
if this is too much I can just keep the ones that are not directly available without `builtins`, which is like 1/4 of those

Reply:
Technically the remaining ones are available with a `__` prefix iirc.  ex: `builtins.readDir` is accessible via `__readDir`

Reply:
I added all the builtins/primops I was able to find